### DISCUSSION TITLE
undeclared identifier

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [pydantic/pydantic#7801] [Question]
The compilation (with mypyc on Windows) of this pydantic V2 code:

```
from typing import Optional
from pydantic import BaseModel

class ThisRecord(BaseModel):
    type: str
    title: str
    status: Optional[int] = None
    detail: Optional[str] = None
    instance: Optional[str] = None
```
... generates the errors:

```
build\__native.c(191): error C2065: 'CPyModule_pydantic___main': undeclared identifier
build\__native.c(191): warning C4047: '=': 'PyObject *' differs in levels of indirection from 'int'
```

Are these pydantic related?

**Comment by dineshbvadhia** [comment]
More info on creating a c-extension of the above code:

Both Pydantic V1 and V2 give the identical error.

Replacing `status: Optional[int] = None` with `status: Optional[str] = None` gives the same error.

Commenting out the `Optional` lines - status, detail and instance - gives the same error.

What am I missing?

**Reply by guywilsonjr** [reply]
did you figure out the answer here? I'm stuck seeing the same error
